recipes/agent.rb

```diff
@@ -1,52 +1,41 @@
 include_recipe "zabbix::agent_#{node['zabbix']['agent']['install_method']}"
 include_recipe 'zabbix::agent_common'
 
-# Install configuration
-template 'zabbix_agentd.conf' do
-  path node['zabbix']['agent']['config_file']
-  source 'zabbix_agentd.conf.erb'
-  unless node['platform_family'] == 'windows'
+# Install configuration and optional additional agent config file containing UserParameter(s)
+case node['platform_family']
+when 'windows'
+  template 'zabbix_agentd.conf' do
+    path node['zabbix']['agent']['config_file']
+    source 'zabbix_agentd.conf.erb'
+    notifies :enable, 'service[Zabbix Agent]'
+    notifies :restart, 'service[Zabbix Agent]'
+    notifies :run, 'execute[config_firewall]'
+  end
+  template 'user_params.conf' do
+    path node['zabbix']['agent']['userparams_config_file']
+    source 'user_params.conf.erb'
+    notifies :enable, 'service[Zabbix Agent]'
+    notifies :restart, 'service[Zabbix Agent]'
+    notifies :run, 'execute[config_firewall]'
```

Comment:
See my earlier comment on use of template variables. It's very confusing to trace why re-templating these files would possibly cause us to reconfigure the firewall. In order to figure out why we're doing that, you have to

1. Read `recipes/_agent_common_service.rb` and understand what `execute[config_firewall]` is doing.
1. Read `recipes/agent.rb` and understand these templates.
1. Go read [`templates/default/user_params.conf.erb`](https://github.com/paxos-bankchain/zabbix/blob/master/templates/default/user_params.conf.erb) and figure out why that might be restarting the firewall.

Upon doing so, we would realize that the `zabbix_agentd.conf.erb` template does not use any ports in the `execute[config_file]` resource. Why are we re-executing the firewall config, then?

Reply:
Changed the execute to not executing firewall config if exist.